File: solucoes/2020/nivel jr/fase 1/py/escher.py
Q = int(input()) 
E = list(map(int, input().split()))
nums = []

for i in range(Q-1):
    if((E[i+1] + E[Q-i-2]) == (E[i] + E[Q-i-1])):
        nums.append("S")
    else:
        nums.append("N")

if "N" in nums:
    print("N")
else:
    print("S")

File: solucoes/2020/nivel jr/fase 1/js/escher.js
var Q 
var E
var S = []
var nums = []

scanf("%d", "Q")

for(var i = 0; i < Q; i++){
    scanf("%d", "E")
    S.push(E)
}

for(var i = 0; i < Q / 2; i++){
    if((S[i+1] + S[Q-i-2]) === (S[i] + S[Q-i-1])){
        nums.push("S")
    }
    else {
        nums.push("N")
    }
}


if(nums.includes("N")){
    printf("%s\n", "N")
}
else{
    printf("%s\n", "S")
}